distribution_metadata.c

```diff
@@ -778,8 +827,10 @@ LockShard(int64 shardId, LOCKMODE lockMode)
 	memset(&lockTag, 0, sizeof(LOCKTAG));
 
 	Assert(lockMode == ExclusiveLock || lockMode == ShareLock);
+	Assert(shardLockType = !SHARD_LOCK_INVALID_FIRST);
```

Comment:
Well that's rather bizarre. I must have transposed characters at some point because I'd never have written this. Fixing.